### PR TITLE
feat: `watchPendingTransactions` + `useWatchPendingTransactions`

### PR DESCRIPTION
**Pull request by jxom**
Co-Authored-By: wslyvh <wslyvh@users.noreply.github.com>

Original PR: #1536 

## Description

Add a `watchPendingTransactions` action that subscribes to `pending` transactions as described in #1535 
I still need to add documentation, but wanted to create the PR for feedback

## Additional Information

- [x] I read the [contributing docs](/wagmi-dev/wagmi/blob/main/.github/CONTRIBUTING.md) (if this is your first contribution)

Your ENS/address:
wslyvh.eth

### CODE REVIEW

File: packages/react/src/hooks/transactions/useWatchPendingTransactions.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  chainId?: number
+  /** Subscribe to changes */
+  enabled?: boolean
+  /** Function fires when a pending transaction enters the mempool. */
+  listener: (transaction: Transaction) => void
+}
+
+export function useWatchPendingTransactions({
+  chainId: chainId_,
+  enabled = true,
```
Comment (wslyvh) [inline packages/react/src/hooks/transactions/useWatchPendingTransactions.ts:18]: this option seems redundant. You either subscribe to to events by using the hook or you don't.

**Review by wslyvh — COMMENTED** [review]
Curious to hear your thoughts around the use of hooks vs. actions? Are hooks separated just to follow hooks rules or is there any other benefit from separating the 2?

For most cases there's quite some overlap, but often (like this PR/pendingTransactions) it's not really using/referencing the same code.

File: packages/react/src/hooks/transactions/useWatchPendingTransactions.ts (hunk 1 above)
Comment (jxom) [reply packages/react/src/hooks/transactions/useWatchPendingTransactions.ts:18]: There may be a case where a consumer wants to disable subscriptions based on a depdendant state variable. wdyt?

**Comment by jxom** [comment]
Generally would like the Hooks to reach parity with the core Actions. Reason why they don't share the same code is because we'd like to leverage the React lifecycle rather than relying on Zustand's reactivity (in core).

File: packages/react/src/hooks/transactions/useWatchPendingTransactions.ts (hunk 1 above)
Comment (wslyvh) [reply packages/react/src/hooks/transactions/useWatchPendingTransactions.ts:18]: Sure, yeah. That makes sense

**Comment by wslyvh** [comment]
Got it. Thanks! 

LGTM then :)

File: .changeset/large-steaks-kick.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@wagmi/core': minor
+---
+
+Add watchPendingTransactions
```
Comment (tmm) [inline .changeset/large-steaks-kick.md:5]: Thoughts on this just being a `patch`?

```suggestion
Added `watchPendingTransactions`
```

File: docs/pages/core/actions/_meta.en-US.json [hunk 3]
```diff
@@ -35,6 +35,7 @@
   "watchReadContract": "watchReadContract",
   "watchReadContracts": "watchReadContracts",
   "watchSigner": "watchSigner",
+  "watchPendingTransactions": "watchPendingTransactions",
```
Comment (tmm) [inline docs/pages/core/actions/_meta.en-US.json:38]: should make this alphabetized

File: docs/pages/core/actions/watchPendingTransactions.en-US.mdx [hunk 4]
```diff
@@ -0,0 +1,48 @@
+---
+title: 'watchPendingTransactions'
+description: 'Action for subscribing to pending transactions.'
+---
+
+import { Callout } from 'nextra-theme-docs'
+
+# watchPendingTransactions
+
+Action for subscribing to pending transactions in the Provider's transaction mempool. [What is a mempool?](https://www.alchemy.com/overviews/what-is-a-mempool)
```
Comment (tmm) [inline docs/pages/core/actions/watchPendingTransactions.en-US.mdx:10]:
```suggestion
Action for subscribing to pending transactions in the Provider's transaction mempool. See ["What is a mempool?"](https://www.alchemy.com/overviews/what-is-a-mempool) for more info.
```

File: docs/pages/react/hooks/useWatchPendingTransactions.en-US.mdx [hunk 5]
```diff
@@ -0,0 +1,71 @@
+---
+title: 'useWatchPendingTransactions'
+description: 'Action for subscribing to pending transactions.'
+---
+
+import { Callout } from 'nextra-theme-docs'
+
+# useWatchPendingTransactions
+
+Hook for subscribing to pending transactions in the Provider's transaction mempool. [What is a mempool?](https://www.alchemy.com/overviews/what-is-a-mempool)
```
Comment (tmm) [inline docs/pages/react/hooks/useWatchPendingTransactions.en-US.mdx:10]:
```suggestion
Hook for subscribing to pending transactions in the Provider's transaction [mempool](https://www.alchemy.com/overviews/what-is-a-mempool).
```

**Review by tmm — COMMENTED** [review]
Couple suggestions, but overall lgtm

**Comment by tmm** [comment]
Needs a changeset for `wagmi` too!

**Pull request merged**